File: ca.cpp
#include <iostream>
#include <iomanip>
#include <vector>

class Stock {
protected:
    double cp;
    int noOfShares;

public:
    void setStock(double price, int shares) {
        cp = price;
        noOfShares = shares;
    }
};

class PfVal : public Stock {
public:
    double calculatePfVal() {
        return cp * noOfShares;
    }
};

class Profit : public Stock {
private:
    double bp;

public:
    void setbp(double price) {
        bp = price;
    }

    double calProf() {
        return (cp - bp) * noOfShares;
    }
};

std::vector<double> calPtfAndPf(double cp, double bp, int noOfShares) {
    PfVal portfolio;
    Profit profit;

    // Set values
    portfolio.setStock(cp, noOfShares);
    profit.setStock(cp, noOfShares);
    profit.setbp(bp);

    // Calculate values
    double PfVal = portfolio.calculatePfVal();
    double profitValue = profit.calProf();

    return {PfVal, profitValue};
}

void readHiddenTestInput(double &cp, double &bp, int &noOfShares) {
    std::cin >> cp >> bp >> noOfShares;
}

int main() {
    // Input
    double cp, bp;
    int noOfShares;

    // Read input from the hidden test case
    readHiddenTestInput(cp, bp, noOfShares);

    // Run test case
    std::vector<double> result = calPtfAndPf(cp, bp, noOfShares);

    // Output
    std::cout << std::fixed << std::setprecision(2);
    std::cout << "Portfolio Value: " << result[0] << std::endl;
    std::cout << "Profit: " << result[1] << std::endl;

    return 0;
}

File: ca.py
class Stock:
    def __init__(self, cp=0.0, no_of_shares=0):
        self.cp = cp
        self.no_of_shares = no_of_shares

    def set_stock(self, cp, no_of_shares):
        self.cp = cp
        self.no_of_shares = no_of_shares

    def calculate_portfolio_value(self):
        return self.cp * self.no_of_shares


class Profit(Stock):
    def __init__(self, bp=0.0):
        super().__init__()
        self.bp = bp

    def set_bp(self, bp):
        self.bp = bp

    def calculate_profit(self):
        return (self.cp - self.bp) * self.no_of_shares


def calculate_portfolio_and_profit(cp, bp, no_of_shares):
    stock = Stock()
    stock.set_stock(cp, no_of_shares)

    profit = Profit()
    profit.set_stock(cp, no_of_shares)
    profit.set_bp(bp)

    portfolio_value = stock.calculate_portfolio_value()
    profit_value = profit.calculate_profit()

    return portfolio_value, profit_value

if __name__ == "__main__":
    cp = float(input())
    bp = float(input())
    NoOfShare = int(input())

    result = calculate_portfolio_and_profit(cp, bp, NoOfShare)

    print(f"Portfolio Value: {result[0]:.2f}")
    print(f"Profit: {result[1]:.2f}")
